src/content/docs/patch-signing.mdx

```diff
@@ -0,0 +1,142 @@
+released application. The Shorebird updater will enforce that only patches
+signed with a corresponding private key will be allowed to load for applications
+that include a public key.
+
+Second, when you build your patch with `shorebird patch`, you can pass
+`--private-key-path` to have Shorebird sign your patch with your private key.
+This is required if you created your release with a public key included.
+
+There are no required changes to your code and you can add or remove this
+signing requirement at any time by simply making a new release of your
+application.
+
+## Adding patch signing to your application
+
+To start, you will need an RSA key pair. Shorebird has only tested with a
+limited set of signing algorithms at this time. Please contact us if you have
+other requirements—we'd be happy to work with you to support your needs.
+
+The `shorebird` tool expects to be able to read private and public keys
+from `.pem` files on disk. If you need other ways of accessing your key material
+(or signing via a cloud signing service) please contact us. We'd be happy to
+work with you to support such.
+
+### Generate keys
+
+If you do not already have an RSA key pair you'd like to use, you can generate a
+pair with `openssl`:
+
+```sh
+# Generate a key pair
+openssl genrsa -out private.pem 2048
+
+# Extract the public key
+openssl rsa -in private.pem -outform PEM -pubout -out public.pem
+```
+
+The above operation will generate a public/private key pair in `private.pem` and
+a public key in `public.pem`. The file containing the private key should be
+stored securely and kept secret. While the private key is not itself sufficient
+to make an update to your application (someone would also need access to your
+Shorebird credentials), it should not be checked into public source control.
+
+### Create a release containing the public key
+
+To create a release that requires signed patches, run the following command:
+
+```sh
+shorebird release android --public-key-path /path/to/public.pem
+```
+
+This will include the public key in the release artifact produced by `shorebird
+release` and cause the released app to require signed patches.
+
+:::note
+We're using Android to demo, but this will work on any platform Shorebird
+supports.
+:::
+
+### Create a signed patch
+
+To create a signed patch, run the following command:
+
+```sh
+shorebird patch android --private-key-path /path/to/private.pem
+```
+
+This tells shorebird to sign the patch with the private key you provide.
+
+:::note
+Storing private keys on disk is not best-practice, and we recognize
+this. This was built as a demonstration of the system, we look forward to
+working with customers to integrate Shorebird to using the signing services and
+key storage methods they are used to.
+:::
+
+### Test it out
+
+You can verify that the patch is properly signed using `shorebird preview`. On
+the first launch, you should see something like the following in your app logs:
```

Comment:
Are the logs missing?

Reply:
Ha, good catch. The Google-Doc-to-markdown converter I used turns out to not be very good 😬

Reply:
Fixed